crates/plex-api/tests/transcode.rs

```diff
@@ -0,0 +1,1784 @@
+                    .unwrap()
+                    .text()
+                    .await
+                    .unwrap();
+
+                let _media_playlist = MediaPlaylist::try_from(text.as_str()).unwrap();
+            } else {
+                panic!("Expected a media stream");
+            }
+
+            cancel(&server, session).await;
+        }
+
+        #[plex_api_test_helper::online_test]
+        async fn hls_transcode_copy(#[future] server: Server) {
+            let server = generify(server.await).await;
+
+            let movie: Movie = server.item_by_id(55).await.unwrap().try_into().unwrap();
+            assert_eq!(movie.title(), "Big Buck Bunny");
+
+            let media = &movie.media()[0];
+            let part = &media.parts()[0];
+
+            let session = movie
+                .create_streaming_session(
+                    part,
+                    Protocol::Hls,
+                    // These settings should allow for direct streaming of the video
+                    // and audio.
+                    VideoTranscodeOptions {
+                        bitrate: 200000000,
+                        width: 1280,
+                        height: 720,
+                        video_codecs: vec![VideoCodec::H264],
+                        audio_codecs: vec![AudioCodec::Aac],
+                        ..Default::default()
+                    },
+                )
+                .await
+                .unwrap();
+
+            verify_session(
+                &session,
+                Protocol::Hls,
+                ContainerFormat::MpegTs,
+                Some((Decision::Copy, AudioCodec::Aac)),
+                Some((Decision::Copy, VideoCodec::H264)),
+            );
+
+            let mut buf = Vec::<u8>::new();
+            session.download(&mut buf).await.unwrap();
+            let index = std::str::from_utf8(&buf).unwrap();
+            let playlist = MasterPlaylist::try_from(index).unwrap();
+            if let VariantStream::ExtXStreamInf { uri, .. } = &playlist.variant_streams[0] {
+                let path = format!("/video/:/transcode/universal/{uri}");
+                let text = server
+                    .client()
+                    .get(path)
+                    .send()
+                    .await
+                    .unwrap()
+                    .text()
+                    .await
+                    .unwrap();
+
+                let _media_playlist = MediaPlaylist::try_from(text.as_str()).unwrap();
+            } else {
+                panic!("Expected a media stream");
+            }
+
+            cancel(&server, session).await;
+        }
+
+        #[plex_api_test_helper::online_test_claimed_server]
+        async fn offline_transcode(#[future] server_claimed: Server) {
+            let server = generify(server_claimed.await).await;
+
+            if !server
+                .media_container
+                .owner_features
+                .contains(&ServerFeature::SyncV3)
```

Comment:
I just was super-curious how do you check for PlexPass :)

Are you sure SyncV3 fits here? Both of my users (free and plex-pass) have it. On the other hand, `HardwareTranscoding` and `Hwtranscode` present only for PlexPass user.

P.S. I didn't look in the `owner_features`, I only checked the features returned within the https://plex.tv/api/v2/user response.

Reply:
Ah maybe that's not right then, unclaimed servers didn't appear to have SyncV3 so I assumed that was it. I can change it to HardwareTranscode if that is a better fit.

Reply:
Just for fun, I compared the features :)

The ones below are present in a non-plexpass user but missed in an unclaimed server:
```
{'AmazonLoopDebug', 'Metadata', 'Singleitemsharing', 'Vod', 'SharedRadio', 'AvodNewMedia', 'GlobalContinueWatching', 'LetsEncrypt', 'ServerManager', 'SaveToLibrary', 'AdCountdownTimer', 'SubtitlesOnDemand', 'TypeFirst', 'ArtistTv', 'CameraUpload', 'MusicPreview', 'SyncV3', 'JoinDiscoverTogetherAsFriend', 'ClientRadioStations', 'Music', 'TunefindClients', 'DownloadCertificates', 'Loudness'}
```

And those are present in a plexpass user but missed in the free one:
```
{'Cloudsync', 'JoinDiscoverTogether', 'Sync', 'SessionKick', 'PremiumDashboard', 'GrandfatherSync', 'VolumeLeveling', 'ItemClusters', 'HardwareTranscoding', 'Webhooks', 'SweetFades', 'Dvr', 'PhotosV6Edit', 'WatchlistRss', 'Hwtranscode', 'Lyrics', 'PhotosV6TvAlbums', 'DvrBlockUnsupportedCountries', 'MusicVideos', 'ContentFilter', 'WebDesktopGracenoteBanner', 'EpgRecentChannels', 'PremiumMusicMetadata', 'SessionBandwidthRestrictions', 'Trailers', 'ImaggaV2', 'MusicAnalysis', 'PhotosV5', 'PhotoAutotags', 'DownloadsGating', 'AllowDvr', 'SigninNotification', 'AlbumTypes', 'BoostVoices', 'SleepTimer', 'SilenceRemoval', 'Nominatim', 'Pass', 'Visualizers', 'IntroMarkers'}
```

I guess `HardwareTranscoding` is indeed the best option for transcoding-related logic.